docs/open_source/releases-and-branches.md

```diff
@@ -0,0 +1,73 @@
+# Releases and Branches
+
+This document describes the branching and release strategy we use for MDC Web, and how it affects dependent repositories
+we maintain, namely the [MDC Web Catalog](https://github.com/material-components/material-components-web-catalog) and
+[MDC React](https://github.com/material-components/material-components-web-react).
+
+For an explanation of the steps needed to cut a release of MDC Web, see [Release Process](./release-process.md).
+
+## MDC Web Release Schedule and Versioning
+
+MDC Web follows a 2-week release cycle. We expect to have one release per month containing breaking changes, to coincide
+with the [Material Design Roadmap](https://github.com/material-components/material-components/blob/develop/ROADMAP.md).
+Any other interim releases will be patch releases including any non-breaking / non-feature changes that can be
+cherry-picked from master in a straightforward manner.
+
+This is a change from our earlier process, where we simply tagged a release with whatever has been merged to master
+within the two-week cycle, and would rev the version number accordingly based on whether there were breaking changes.
+
+## Release Types
+
+### Breaking-change Releases
+
+These are currently our minor releases, e.g. 0.35.0, 0.36.0, etc. These follow the currently-outlined
+[Release Process](./release-process.md) and are performed against `master`.
+
+### Pre-releases
+
+MDC Web Catalog and MDC React both depend on MDC Web via npm dependencies. In order to be able to integrate
+breaking changes for the upcoming release ahead of time - preferably while the necessary changes are fresh in someone's
+mind - we will cut pre-releases of MDC Web when breaking changes are merged which warrant updates in the other
+repositories. These will be cut on an as-needed basis, with no set schedule.
+
+The Catalog and React repositories will each have a `next` branch, which will be updated to point to the pre-release.
+Pull requests should be filed against this branch for changes needed for the upcoming MDC Web release.
+
+Later, when the new version of MDC Web is released, the `next` branch will again be updated to point to the new release,
+and we will open a PR to squash & merge its contents onto `master`. Once this is done, the `next` branch will be reset
```

Comment:
"squash & merge" - so are you saying we will be losing all the "next" branch commits?

Reply:
I was mostly thinking to preserve the normal PR+review+squash/merge process for master rather than needing to temporarily override it. Otherwise we would need to temporarily disable some of the master branch protections in order to cherry-pick and push commits into master. On the other hand, this does mean it'd need to get reviewed quickly, preferably, but everything on `next` should have already been reviewed once in detail. It also hopefully means only resolving conflicts once, vs. potentially multiple times for cherry-picks.

Do you think there's a strong argument for keeping the commits separate?

Reply:
I'm saying both.

Each change targeting `next` should go through the PR process and be squashed+merged into it. This may result in multiple commits to `next`.

When it's time to bring everything over into `master`, send a PR and squash+merge `next` into `master` (i.e. you end up with one "update for breaking changes" commit in master). The alternative is temporarily unprotecting master to just push either a squashed commit or cherry-picks straight through, but I'd prefer the PR approach as one final validation in case anything skewed between `master` and `next` since `next` was branched.

Reply:
If we squash everything from next --> master, we would lose those commits for the Changelog, no?

Reply:
That's a good point for MDC React, if that will involve individual breaking changes to it as well. I already followed the squash+merge process once for Catalog, where changelog isn't a consideration.